go/enclave/evm/evm_facade.go

```diff
@@ -40,12 +44,16 @@ func ExecuteTransactions(txs []*common.L2Tx, s *state.StateDB, header *common.He
 	return result
 }
 
-func executeTransaction(s *state.StateDB, cc *params.ChainConfig, chain *ObscuroChainContext, gp *gethcore.GasPool, header *common.Header, t *common.L2Tx, usedGas *uint64, vmCfg vm.Config, tCount int) (*types.Receipt, error) {
+func executeTransaction(s *state.StateDB, cc *params.ChainConfig, chain *ObscuroChainContext, gp *gethcore.GasPool, header *types.Header, t *common.L2Tx, usedGas *uint64, vmCfg vm.Config, tCount int) (*types.Receipt, error) {
 	s.Prepare(t.Hash(), tCount)
 	snap := s.Snapshot()
 
+	before := header.MixDigest
```

Comment:
Could spell it out a bit clearer here for people not familiar, since this is all a bit magic.

Could maybe name `before` something like `rollupRandomness` or maybe just a comment to say something along the lines of 
`// for each tx execution we provide unique random bytes, making them available through the header.MixDigest property`

And a comment for line 56 to say `setting MixDigest back to rollup-level random data for the next transaction` or something.